src/queue-list/laboratory-patient-list.component.tsx

```diff
@@ -13,111 +14,222 @@ import {
+import {
+  formatWaitTime,
+  getTagColor,
+  trimVisitNumber,
+} from "../utils/functions";
+
+// type FilterProps = {
+//   rowIds: Array<string>;
+//   headers: Array<DataTableHeader>;
+//   cellsById: any;
+//   inputValue: string;
+//   getCellId: (row, key) => string;
+// };
 
-interface LaboratoryPatientListProps {
-  searchTerm: string;
-  location: string;
-  status: string;
-}
+interface LaboratoryPatientListProps {}
 
-const LaboratoryPatientList: React.FC<LaboratoryPatientListProps> = ({
-  searchTerm,
-  location,
-  status,
-}) => {
+const LaboratoryPatientList: React.FC<LaboratoryPatientListProps> = () => {
   const { t } = useTranslation();
+  const session = useSession();
+
+  const { patientQueueEntries, isLoading } = usePatientQueuesList(
+    session?.sessionLocation?.uuid,
+    status
+  );
+
+  const pageSizes = [10, 20, 30, 40, 50];
   const [page, setPage] = useState(1);
-  const [pageSize, setPageSize] = useState(10);
+  const [currentPageSize, setPageSize] = useState(10);
   const [nextOffSet, setNextOffSet] = useState(0);
-  const [laboratoryQueue, setLaboratoryQueue] = useState([]);
-  const totalOrders = 0;
+
+  const {
+    goTo,
+    results: paginatedQueueEntries,
+    currentPage,
+  } = usePagination(patientQueueEntries, currentPageSize);
 
   let columns = [
-    { header: t("visitId", "Visit ID"), key: "visitId" },
-    { header: t("names", "Names"), key: "names" },
-    { header: t("age", "Age"), key: "age" },
-    { header: t("orderedFrom", "Ordered from"), key: "orderedFrom" },
-    { header: t("waitingTime", "Waiting time"), key: "waitingTime" },
-    { header: t("testsOrdered", "Tests ordered"), key: "testsOrdered" },
+    { id: 0, header: t("visitId", "Visit ID"), key: "visitId" },
+    { id: 1, header: t("names", "Names"), key: "names" },
+    { id: 2, header: t("age", "Age"), key: "age" },
+    { id: 3, header: t("orderedFrom", "Ordered from"), key: "orderedFrom" },
+    { id: 4, header: t("waitingTime", "Waiting time"), key: "waitingTime" },
   ];
 
-  useEffect(() => {
-    setPage(1);
-    setNextOffSet(0);
-  }, [searchTerm]);
+  const tableRows = useMemo(() => {
+    return paginatedQueueEntries?.map((entry) => ({
+      ...entry,
+      visitId: {
+        content: <span>{trimVisitNumber(entry.visitNumber)}</span>,
+      },
+      names: {
+        content: <span>{entry.name}</span>,
+      },
+      age: {
+        content: <span>{age(entry.patientDob)}</span>,
+      },
+      orderedFrom: {
+        content: <span>{entry.locationFromName}</span>,
+      },
+      waitingTime: {
+        content: (
+          <Tag>
+            <span
+              className={styles.statusContainer}
+              style={{ color: `${getTagColor(entry.waitTime)}` }}
+            >
+              {formatWaitTime(entry.waitTime, t)}
+            </span>
+          </Tag>
+        ),
+      },
+    }));
+  }, [paginatedQueueEntries, t]);
+
+  // const handleFilter = ({
```

Comment:
will this be needed later on? if not then we shoild delete them as well

Reply:
will be needed